zipline/assets/assets.py

```diff
@@ -703,6 +704,10 @@ def lookup_symbol(self, symbol, as_of_date, fuzzy=False):
             there are multiple candidates for the given ``symbol`` on the
             ``as_of_date``.
         """
+        if symbol is None:
+            raise TypeError("Cannot lookup symbol of type NoneType for "
+                            "as of date %s" % as_of_date)
```

Comment:
I can buy that.  What do you think about shortening it to mention just "None" instead of "NoneType"?  I find that clearer, but maybe everyone is used to "AttributeError: Instance of NoneType has no attribute..." ?